### PR TITLE
Replace the typescript compiler with @babel/preset-typescript

### LINKED ISSUE #6938: Use Babel to compile core files instead of tsc
Since we want to reliably know which features are / are not compiled and we're already running Babel on most of them as a post-processing step it makes sense to use `@babel/preset-typescript` instead of `tsc` this will also avoid Typescript specific helpers (__importStar etc) and use the Babel ones which are already included in the bundles because user code is compiled using Babel.

### PR DESCRIPTION
**Pull request by lfades**
This change only affects the core files.

Closes #6938

### CODE REVIEW

File: packages/next/taskfile.js [hunk 1]
```diff
@@ -24,6 +27,23 @@ const babelOpts = {
...
+    ['@babel/preset-env', {
+      modules: 'commonjs',
+      targets: {
+        node: '8.3'
+      },
+      exclude: ['transform-typeof-symbol']
+    }]
+  ],
+  plugins: [
+    'babel-plugin-dynamic-import-node',
```
Comment (timneutkens) [inline packages/next/taskfile.js]: This looks incorrect, it should not compile `import()`

Comment (lfades) [reply packages/next/taskfile.js]: @timneutkens That's only for the server files used by the core, and Node can't handle that.

Comment (timneutkens) [reply packages/next/taskfile.js]: But it wasn't i think

Comment (lfades) [reply packages/next/taskfile.js]: @timneutkens I removed `babel-plugin-dynamic-import-node` from everywhere except the `bin` task 👍

**Review by timneutkens — APPROVED** [review]

**Pull request merged**